### PR TITLE
feat: Updated `TF Contribution Guide` and aligned with `Bicep Contribution Guide`

### PR DESCRIPTION
**Pull request by segraef**
# Overview/Summary

Updated `TF Contribution Guide` and aligned with `Bicep Contribution Guide`.
Also updated TF specs titles

image
image

## As part of this Pull Request I have

- [x] Read the Contribution Guide and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Verified-Modules/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Verified-Modules/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Verified-Modules/)
- [x] Ensured PR tests are passing
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Review by mbilalamjad — DISMISSED** [review]

File: docs/content/contributing/terraform/terraform-contribution-flow/_index.md [hunk 1]
```diff
@@ -105,7 +105,7 @@ To do so, simply navigate to your desired repository, select the `'Fork'` button
 
 {{< hint type=note >}}
 
-If the module repository you want to contribute to is not yet available, please get in touch with the respective module owner which can be tracked in the [Terraform Resource Modules index](https://azure.github.io/Azure-Verified-Modules/indexes/terraform/tf-resource-modules/) see `PrimaryModuleOwnerGHHandle` column.
+If the module repository you want to contribute to is not yet available, please get in touch with the respective module owner which can be tracked in the [Terraform Resource Modules index](/Azure-Verified-Modules/indexes/terraform/tf-resource-modules/) see `PrimaryModuleOwnerGHHandle` column.
```
Comment (arnoldna) [inline docs/content/contributing/terraform/terraform-contribution-flow/_index.md:108]:
```suggestion
If the module repository you want to contribute to is not yet available, please contact the respective module owner via the resource module index.  [Terraform Resource Modules index](/Azure-Verified-Modules/indexes/terraform/tf-resource-modules/) see `PrimaryModuleOwnerGHHandle` column.
```

File: docs/content/contributing/terraform/prerequisites.md [hunk 2]
```diff
@@ -7,12 +7,9 @@ geekdocAnchor: true
...
 {{< toc >}}
 
-<!--
-NOTE:
+## GitHub Account Link and Access
 
-Removed ## Recommended Learning since advanced (not basic) knowledge is required around Git, TF and Docker, it is considered a hard requirement. Someone just looking at the Terraform Docs or code and trying to figure out how to contribute will not be successful. Contributors need to have a good understanding and holistic view of all tools and how to use them.
-
--->
+To contribute to this project, you need to have a GitHub account which is [linked](https://repos.opensource.microsoft.com/link) to your Microsoft FTE corporate identity account and be a member of the [Azure](https://repos.opensource.microsoft.com/orgs/Azure) organisation.
```
Comment (arnoldna) [inline docs/content/contributing/terraform/prerequisites.md:12]:
```suggestion
To contribute to this project, you need to have a personal GitHub account which is [linked](https://repos.opensource.microsoft.com/link) to your Microsoft corporate identity and be a member of the [Azure](https://repos.opensource.microsoft.com/orgs/Azure) organization.
```

File: docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md [hunk 3]
```diff
@@ -31,25 +35,88 @@ Make sure module authors/contributors tested their module in their environment b
...
+
+Examples:
 
-- `@Azure/avm-res-<RP>-<modulename>-module-owners-tf`
-- `@Azure/avm-res-<RP>-<modulename>-module-contributors-tf`
+<!-- TODO: Clarify parent teams with the team -->
 
-2. Create the module repostory using [terraform-azuremrm-avm-template](https://github.com/Azure/terraform-azurerm-avm-template) in the `Azure` organisation with the following [details (internal only)](https://dev.azure.com/CSUSolEng/Azure%20Verified%20Modules/_wiki/wikis/AVM%20Internal%20Wiki/333/-TF-Create-repository-in-Github-Azure-org-and-conduct-business-review). You will then have to complete the configuration of your repository and start an [internal business review](https://dev.azure.com/CSUSolEng/Azure%20Verified%20Modules/_wiki/wikis/AVM%20Internal%20Wiki/333/-TF-Create-repository-in-Github-Azure-org-and-conduct-business-review?anchor=conduct-initial-repo-configuration-and-trigger-business-review).
+- `avm-res-compute-virtualmachine-module-owners-tf` and added `avm-technical-reviewers-tf` as parent.
+- `avm-res-compute-virtualmachine-module-contributors-tf` and added `avm-technical-reviewers-tf` as parent.
```
Comment (segraef) [inline docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md:102]: @matt-FFFFFF and @mbilalamjad this needs to be decided, if we go with parent teams we need to decide on the names. Parent teams are required for the teams permissions inheritance housekeeping.

Comment (matt-FFFFFF) [reply docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md:102]: Why do we want them parented?

File: docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md [hunk 4]
```diff
@@ -165,6 +256,19 @@ When an AVM Module is published to the HashiCorp Registry, it **MUST** follow th
...
 
+#### Updated module version and bublish new version
+
+Once a module was updated and is ready to be published, follow the below steps to resync your repository to publish the latest module version to the HashiCorp Registry.
+
+1. Make sure your module version is reflected in the `locals.version.tf.json` file.
+2. Publish a new version by creating a new tag and release on GitHub as describe above.
+3. Sign in to the [HashiCorp Registry](https://registry.terraform.io/) using GitHub.
+4. Visit your module on https://registry.terraform.io/modules/Azure/.
+5. Resync the module by selecting the `Resync Module` button in the top right corner.
```
Comment (matt-FFFFFF) [inline docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md:267]: This step is not necessary

File: docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md (hunk 3 above)
Comment (segraef) [reply docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md]: Removed.

File: docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md (hunk 4 above)
Comment (segraef) [reply docs/content/contributing/terraform/terraform-contribution-flow/owner-contribution-flow.md]: Updated.

**Review by matt-FFFFFF — DISMISSED** [review]

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**